components/skeleton/skeleton-styles.js

```diff
@@ -0,0 +1,30 @@
+import { css } from 'lit-element/lit-element.js';
+
+export const skeletonStyles = css`
+	@keyframes loadingPulse {
+		0% { background-color: var(--d2l-color-sylvite); }
+		50% { background-color: var(--d2l-color-regolith); }
+		75% { background-color: var(--d2l-color-sylvite); }
+		100% { background-color: var(--d2l-color-sylvite); }
+	}
+
+	:host([skeleton]) .skeletize::before {
+		content: "";
+		position: absolute;
+		top: 0;
+		bottom: 0;
+		right: 0;
+		left: 0;
+		border-radius: 0.2rem;
+		animation: loadingPulse 1.8s linear infinite;
+		background-color: var(--d2l-color-sylvite);
+		z-index: 2000;
+	}
+
```

Comment:
Possibly? a little hacky but this style sets the component being styled to `position: relative` to anchor the absolutely positioned `::before` selector. Not sure if that could cause issues in other places where position is being used for other things, but it works for us in FACE for now. Also this style includes other overrides that might 'bleed' out of the skeleton like borders, box-shadows etc.

Reply:
This seems reasonable to me.

Yeah I am a little concerned about clobbering styles on the component as you mentioned.  Perhaps those can be addressed on a case by case basis - for example, in such case, the `skeletize` class would perhaps be rendered on another nested element.

What's the rationale for the `z-index: 2000;` value?  I think this could cause the skeleton to be displayed on-top of some other elements that we would not normally expect it to.  For this skeleton approach, I would expect a `z-index` value of something like `1-10` range.

I think it would be appropriate to also define a media query to respect user preference for reduced motion.
```
@media (prefers-reduced-motion: reduce) {
	:host {
		animation: none;
	}
}
```

Reply:
After thinking about this more, I suspect a high z-index is actually what we want to ensure everything gets covered in the skeletized element.  Still worthwhile knowing why `2000` was picked though and if this were merged also applying a brief comment to avoid regressions etc.

Reply:
Yeh- the high z-index was kind of plucked randomly out of thin air because when I was first experimenting with this idea, the skeleton overlay wasn't working for masking things like regular input fields. I meant to go back and dig more into it at some point. I think even just setting a z-index of 1 on the `::before` works around that issue.

But it's things like the z-index that make me feel a little bit uncomfortable about this whole approach - and the use of the ::before pseudo selector in general makes me nervous - mostly because I don't have much experience with using it and coupling it with absolute positioning just seems like it's pushing the boundaries...not sure how consistent browsers are at implementing such things and whether the approach I'm using is a "designed css feature" or just happens to be working in a way that I hoped it would.

Initially I started with the idea of directly rendering an actual `d2l-skeleton` component in each custom element when it was in `shadow` mode and trying to get that `d2l-skeleton` to `fill` the component in such a way that you didn't have to explicitly set dimensions on it. But then it just occurred to me that using `::before` with `content: ""` might be a simple and elegant? way to achieve a similar effect. But I was mighty surprised when it just seemed to work.

Reply:
I think it's a really elegant solution, and one I'm not sure I would have thought of! I'm less worried about the style clobbering and agree we could just tackle any issues that come up with it case-by-case.

The `z-index` is worrying, but I think as long as we tested it out with a bunch of places we also leverage `z-index` (dropdowns, tooltips & dialogs come to mind) hopefully we can sort that out. We've been talking about having some kind of "z-index manager" that helps us keep track of where various things layer in... if things get hairy we could always look into that.

Reply:
I think using a `::before` pseudo-element is elegant and will fit for many use cases. It does remind me of early web dev days, though 😂

Reply:
Also worried about z-index

Reply:
It is nice to have skeletons over things. So when they hide there is not flicker.